doc/register.md

```diff
@@ -5,23 +5,43 @@
 
 @signature `route.register(rule [, defaults])`
 
-Create a url matching rule. Optionally provide defaults that will be applied to the underlying [can-route.data] when the rule matches.
-
-The following sets `route.data.page = "cart"` when the url is `#cart` and
-`route.data.page = "home"` when the url is `#`.
-
-```js
-route.register( "{page}", { page: "home" } );
-```
-
-@param {String} rule the fragment identifier to match.  The fragment identifier should contain characters (a-Z), optionally wrapped in braces ( { } ). Identifiers wrapped in braces are interpreted as being properties on can-route’s map. Examples:
-
-```js
-route.register( "{foo}" );
-route.register( "foo/{bar}" );
-```
-
-@param {Object} [defaults] An object of default values. These defaults are applied to can-route’s map when the route is matched.
-
-@return {Object} The internal route object.  Use values on this object with caution. It is
- subject to change.
+  Create a url matching rule. Optionally provide defaults that will be applied to the underlying [can-route.data] when the rule matches.
+
+  The following sets `route.data.page = "cart"` when the url is `#cart` and `route.data.page = "home"` when the url is `#`.
+
+  ```html
+  <mock-url></mock-url>
+  <script type="module">
+  import "//unpkg.com/mock-url@^5.0.0";
+  import {route} from "can";
+
+  route.register( "{page}", {page: "home"} );
+  
+  route.start();
+  console.log( route.data.page ); // -> "home"
+  route.data.page = "cart";
+  </script>
+  ```
+  @codepen
+
+  @param {String} rule the fragment identifier to match.  The fragment identifier should contain characters (a-Z), optionally wrapped in braces `( { } )`. Identifiers wrapped in braces are interpreted as being properties on can-route’s map. Examples:
```

Comment:
should probably be like "braces (`{ }`)" ... parenthesis should not be within a `<code>` block

Reply:
also


> can-route's map 

should be 

> can-route's [can-route.data].

I'd look for other uses of `map` (which was the old `.data` property).

Reply:
I think the rules here:

> contain characters (a-Z), optionally wrapped in braces `( { } )`

aren't accurate.

Rules like `.register("place/{placeType}-{placeSlug}")` work fine.

We should explain what characters have special meaning.  I think it's `/`, `{`, and `}`, `?`, `#`, `!`, `=`, `[`,`]`, `&`.  Everything else should be fine in the "URL" part.  For example `.register("_||${foo}@^")`.

Within the `{}` we should allow `(a-Z)`, but also probably other typically property identifiers: `$`, `_` and numbers after the first character.  Basically any valid variable name should work.

Reply:
- [x] fixed bracing issues.
- [x] fixed can-route's map